core/src/main/java/com/datastrato/gravitino/catalog/BaseCatalog.java

```diff
@@ -87,7 +87,13 @@ public CatalogOperations ops() {
         if (ops == null) {
           Preconditions.checkArgument(
               entity != null && conf != null, "entity and conf must be set before calling ops()");
-          ops = newOps(conf);
+          CatalogOperations newOps = newOps(conf);
```

Comment:
If the partition API is merged, how do you handle that case?

Reply:
Maybe we will have `TableOperations`. We will create proxy object for it.

Reply:
I think you have to figure out a solution with @mchades ASAP before the interface is settle down.

Reply:
Think twice.  If the type of value is Table, I return a proxy object, too. 
```
  public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
    Object object = plugin.doAs(
        PrincipalUtils.getCurrentPrincipal(),
        () -> method.invoke(ops, args),
        Collections.emptyMap());
    if (object instanceof Table) {
      return createProxy(object, plugin);
    }
    return object;
  }
```

Reply:
I recover the code after discussion with @mchades if we have TableOperation.